pkg/inject/inject.go

```diff
@@ -618,13 +622,14 @@ func (conf *ResourceConfig) injectProxyInit(patch *Patch, saVolumeMount *corev1.
 	capabilities := &corev1.Capabilities{}
 	if conf.pod.spec.Containers != nil && len(conf.pod.spec.Containers) > 0 {
 		if sc := conf.pod.spec.Containers[0].SecurityContext; sc != nil && sc.Capabilities != nil {
-			capabilities = sc.Capabilities
+			capabilities.Add = sc.Capabilities.Add
+			capabilities.Drop = sc.Capabilities.Drop
```

Comment:
curious about this change: do we want to guarantee that we're _only_ copying the `Add` and `Drop` fields? if not, i'd revert this.

Reply:
Yup, just the `Add` and `Drop`, as they are the only exported fields.

Reply:
Also, previously, it ended by copying references to the slides. The end result is that, as  the sidecars' security context is updated, the primary container's is updated too :scream:.